utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,205 @@
+            "HP" => {
+                telecom.use = "home";
+            }
+            "WP" => {
+                telecom.use = "work";
+            }
+        }
+    } else {
+        log:printDebug("Telecom use not available", telecomUse);
+    }
+
+    if telecom.use is string && telecom.value is string{
+        return telecom;
+    } else {
+        log:printDebug("telecom fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA name to FHIR HumanName.
+#
+# + nameElement - C-CDA name element
+# + return - Return FHIR HumanName
+public isolated function mapCcdaNametoFhirName(xml nameElement) returns r4:HumanName? {
+    r4:HumanName name = {};
+    xml familyElement = nameElement/<v3:family|family>;
+    xml givenElement = nameElement/<v3:given|given>;
+
+    string givenName = givenElement.data().trim();
+    string family = familyElement.data().trim();
+
+    if givenName != "" || family != "" {
+        name.given = [givenElement.data()];
+        name.family = familyElement.data();
+        return name;
+    } else {
+        log:printDebug("name fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA code to FHIR CodeableConcept.
+#
+# + codingElement - C-CDA code element
+# + return - Return FHIR CodeableConcept
+public isolated function mapCcdaCodingtoFhirCodeableConcept(xml codingElement) returns r4:CodeableConcept? {
+    r4:Coding? mapCcdaCodingtoFhirCodeResult = mapCcdaCodingtoFhirCode(codingElement);
+    if mapCcdaCodingtoFhirCodeResult is r4:Coding {
+        return {
+            coding: [mapCcdaCodingtoFhirCodeResult]
+        };
+    } else {
+        log:printDebug("codeableConcept not available");
+        return ();
+    }
+}
+
+# Map C-CDA code to FHIR Coding.
+#
+# + codingElement - C-CDA code element
+# + return - Return FHIR Coding
+public isolated function mapCcdaCodingtoFhirCode(xml codingElement) returns r4:Coding? {
+    r4:Coding coding = {};
+    string?|error codeVal = codingElement.code;
+    string?|error systemVal = codingElement.codeSystem;
+
+    if codeVal is string {
+        coding.code = codeVal;
+    } else {
+        log:printDebug("codeVal is not available", codeVal);
+    }
+
+    if systemVal is string {
+        coding.system = systemVal;
+    } else {
+        log:printDebug("systemVal is not available", systemVal);
+    }
+
+    if coding.code is string || coding.system is string {
+        return coding;
+    } else {
+        log:printDebug("coding fields not available");
+        return ();
+    }
+}
+
+# Map C-CDA dateTime to FHIR dateTime.
+#
+# + dateTimeElement - C-CDA dateTime element
+# + return - Return FHIR dateTime
+public isolated function mapCcdaDateTimeToFhirDateTime(xml dateTimeElement) returns r4:dateTime? {
+    string?|error dateTimeVal = dateTimeElement.value;
+    if dateTimeVal is string {
+        int length = dateTimeVal.length();
+        if length == 4 {
+            return dateTimeVal;
+        } else if length == 6 {
```

Comment:
Can just be ifs.

Reply:
The initial code which had set of if checks was changed according to this as a review feedback given. However this was changed to `match` clause in Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/cc257cbf67919d7d86a4e90e88ec03e1250eadcf